File: CS/ViewModels/Base/ChannelViewModel.cs
namespace DXHtmlMessengerSample.ViewModels {
    using DevExpress.DevAV.Chat;
    using DevExpress.DevAV.Chat.Events;
    using DevExpress.Mvvm;
    using DevExpress.Mvvm.POCO;

    public abstract class ChannelViewModel {
        protected ChannelViewModel() {
            Messenger.Default.Register<IChannel>(this, OnConnected);
        }
        protected virtual void OnConnected(IChannel channel) {
            channel.Subscribe(OnChannelEvent);
        }
        public virtual void OnCreate() {
            EnsureDispatcherService();
        }
        public virtual void OnDestroy() {
            Messenger.Default.Unregister<IChannel>(this, OnConnected);
        }
        protected IChannel Channel {
            get;
            private set;
        }
        void OnChannelEvent(ChannelEvent @event) {
            var channelReady = @event as ChannelReadyEvent;
            if(channelReady != null) {
                EnsureDispatcherService();
                Channel = channelReady.Channel;
                OnChannelReady();
            }
        }
        protected virtual void OnChannelReady() { }
        protected IDispatcherService DispatcherService {
            get;
            private set;
        }
        protected IDispatcherService EnsureDispatcherService() {
            return DispatcherService ?? (DispatcherService = this.GetRequiredService<IDispatcherService>());
        }
    }
}

File: VB/ViewModels/Base/ChannelViewModel.vb
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Mvvm
Imports DevExpress.DevAV.Chat.Events
Imports DevExpress.DevAV.Chat

Namespace DXHtmlMessengerSample.ViewModels

    Public MustInherit Class ChannelViewModel
        Protected Sub New()
            Messenger.Default.Register(Of IChannel)(Me, AddressOf OnConnected)
        End Sub
        Protected Overridable Sub OnConnected(ByVal channel As IChannel)
            channel.Subscribe(AddressOf OnChannelEvent)
        End Sub
        Public Overridable Sub OnCreate()
            EnsureDispatcherService()
        End Sub
        Public Overridable Sub OnDestroy()
            Messenger.Default.Unregister(Of IChannel)(Me, AddressOf OnConnected)
        End Sub
        Private privateChannel As IChannel
        Protected Property Channel() As IChannel
            Get
                Return privateChannel
            End Get
            Private Set(ByVal value As IChannel)
                privateChannel = value
            End Set
        End Property
        Sub OnChannelEvent(ByVal [event] As ChannelEvent)
            Dim channelReady = TryCast([event], ChannelReadyEvent)
            If channelReady IsNot Nothing Then
                EnsureDispatcherService()
                Channel = channelReady.Channel
                OnChannelReady()
            End If
        End Sub
        Protected Overridable Sub OnChannelReady()
        End Sub
        Private privateDispatcherService As IDispatcherService
        Protected Property DispatcherService() As IDispatcherService
            Get
                Return privateDispatcherService
            End Get
            Private Set(ByVal value As IDispatcherService)
                privateDispatcherService = value
            End Set
        End Property
        Protected Function EnsureDispatcherService() As IDispatcherService
            If DispatcherService IsNot Nothing Then
                Return DispatcherService
            Else
                DispatcherService = Me.GetRequiredService(Of IDispatcherService)()
                Return DispatcherService
            End If
        End Function
    End Class
End Namespace